src/pyFAI/geometryRefinement.py

```diff
@@ -189,9 +206,23 @@ def __init__(
         self._wavelength_min = 1e-15
         self._wavelength_max = 100.0e-10
 
+    def __copy__(self):
+        """:return: a shallow copy of itself."""
+        new = self.__class__(data=self.data,
+                             detector=self.detector,
+                             calibrant = self.calibrant)
+        for key in self._IMMUTABLE_ATTRS:
+            new.__setattr__(key, self.__getattribute__(key))
```

Comment:
This line ` new.__setattr__(key, self.__getattribute__(key))` requires that attributes exist, for example in the constructor